packages/sources/dxdao/test/integration/adapter.test.ts

```diff
@@ -0,0 +1,77 @@
+import { AdapterRequest, Execute } from '@chainlink/types'
+import * as dxdaoAdapter from "../../src/index"
+import { ethers } from 'ethers'
+
+jest.mock("ethers", () => ({
+  ...(jest.requireActual("ethers")),
+  ethers: {
+    providers: {
+      JsonRpcProvider: function(_: string): ethers.provider.JsonRpcProvider { return {} } 
+    },
+    Contract: function() {
+      return {
+        balanceOf: (____: string) => {
+          return {
+            _hex: '0x0c3f8fe18611f5a7cb'
+          }
+        }
+      }
+    }
+  }
+}))
+
+jest.mock("@chainlink/ea-reference-data-reader", () => ({
+  ...(jest.requireActual("@chainlink/ea-reference-data-reader")),
+  getRpcLatestAnswer: jest.fn().mockReturnValue(2020)
+}))
```

Comment:
Nice job with the mocks! :+1: this will be useful in a lot of other adapters too

Reply:
thanks!